src/server/project.ts

```diff
@@ -1194,6 +1151,25 @@ namespace ts.server {
         }
     }
 
+    function getUnresolvedImports(program: Program, cachedUnresolvedImportsPerFile: Map<ReadonlyArray<string>>): SortedReadonlyArray<string> {
+        const ambientModules = program.getTypeChecker().getAmbientModules().map(mod => stripQuotes(mod.getName()));
+        return toDeduplicatedSortedArray(flatMapToMutable(program.getSourceFiles(), sourceFile =>

```

Comment:
This reverts it back to allocating array?

Reply:
D'oh, #28214